documentation/api/query/v4/pql.markdown

```diff
@@ -0,0 +1,339 @@
+---
+title: "PuppetDB 3.2 » API » v4 » Puppet Query Language (PQL)"
+layout: default
+canonical: "/puppetdb/latest/api/query/v4/pql.html"
+---
+
+[entities]: ./entities.html
+[subquery]: #subqueries
+
+> **Experimental Feature**: This featureset is experimental, and it may be altered or removed in
+> future release.
+
+Puppet Query Language (PQL) is a query language designed with PuppetDB and Puppet data in mind. It
+provides a string based query language as an alternative to the legacy query language PuppetDB has
+always supported.
+
+## Query Structure
+
+A PQL query has the following structure:
+
+    <entity> [<projection>] { <filter> <modifiers> }
+
+As an example, if you wanted to query for `nodes` related data, but only see the `certnames` field
+for each node, with a regex filter across certname:
+
+    nodes[certname] { certname ~ '^web' }
```

Comment:
it would be great to clarify here out of "certname ~ '^web'", if its a filter or a filter and a modifier, it's probably explained further down, but it would be great to clarify the meanings of the term above with an example near their first usage